feat: add missing User struct fields

kduprey opened the pull request:
## Changes

- Add `EnterpriseAccount` type with SAML/OAuth enterprise connection support
- Enhance `User` struct with new fields:
  - `CreatedAt`, `UpdatedAt`, `MFAEnabledAt`, `MFADisabledAt`, `LegalAcceptedAt`
  - `EnterpriseAccounts` for enterprise SSO support
- Update `EmailAddress` with `MatchesSsoConnection` field
- Add shared `Verification` and `LinkedIdentification` types

jfoshee commented on email_address.go:
```diff
@@ -4,17 +4,22 @@ import "encoding/json"
+	Reserved             bool                    `json:"reserved"`
+	Verification         *Verification           `json:"verification"`
+	LinkedTo             []*LinkedIdentification `json:"linked_to"`
+	MatchesSsoConnection bool                    `json:"matches_sso_connection"`
```
Should this be `MatchesSSOConnection` or do we have an existing style exception for this?

kduprey replied:
This is currently the same name as what is in our [main Go repo](https://github.com/clerk/clerk_go/blob/kenton/user-2719-add-user-object-to-sessioncreated-webhook/api/serialize/identification.go#L19-L20).

jfoshee replied:
Granted. But I'm not convinced that because we are not following Go conventions in our internal repo, that we should let that unduly influence our public Go repo. If we decide to rename this later it becomes a breaking change for our customers. But internally we can usually rename more easily. So I think we should try to "get it right the first time" in this repo.

jfoshee commented on enterprise_account.go:
```diff
@@ -0,0 +1,44 @@
+	Name          string  `json:"name"`
+	LogoPublicURL *string `json:"logo_public_url"`
+	// Deprecated: Use Domains instead
+	Domain                           string   `json:"domain,omitempty"`
```
❓  Honest question: as `domain` is already deprecated and isn't already a part of the Go SDK, do we need to include it here?

kduprey replied:
Removed

jfoshee commented on enterprise_account.go:
```diff
@@ -0,0 +1,44 @@
+
+	// SAML Connection specific fields
+	AllowSubdomains   bool `json:"allow_subdomains"`
+	AllowIdpInitiated bool `json:"allow_idp_initiated"`
```
Similar to above, should we make this `AllowIDPInitiated`?

kduprey replied:
This is the same name as what is in our [main Go repo](https://github.com/clerk/clerk_go/blob/kenton/user-2719-add-user-object-to-sessioncreated-webhook/api/serialize/enterprise_account.go#L54-L55).

kduprey replied:
Updated

jfoshee commented on user.go:
```diff
@@ -66,23 +71,44 @@ type Web3Wallet struct {
+	ProviderUserID         *string                `json:"provider_user_id"`
+	LastAuthenticatedAt    *int64                 `json:"last_authenticated_at"`
+	PublicMetadata         json.RawMessage        `json:"public_metadata"`
+	SamlConnection         *samlAccountConnection `json:"saml_connection"`
```
I guess it would be a breaking change to call this type `SAMLConnection`? :-/

kduprey replied:
Yes, due to the change in name. I will still call out that this is the same name as what is in our [main Go repo](https://github.com/clerk/clerk_go/blob/kenton/user-2719-add-user-object-to-sessioncreated-webhook/api/serialize/saml_account.go#L21-L22).

kduprey replied:
Wait, scratch that. We can change the name without breaking change as the field was not present previously.

kduprey replied:
Updated

jfoshee commented on user.go:
```diff
@@ -66,23 +71,44 @@ type Web3Wallet struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+	// Deprecated: Use Domains instead
+	Domain                           string   `json:"domain"`
```
Should `domain` be included, as it is already deprecated?

kduprey replied:
Removed

jfoshee commented on user.go:
```diff
@@ -66,23 +71,44 @@ type Web3Wallet struct {
+	Provider                         string   `json:"provider"`
+	SyncUserAttributes               bool     `json:"sync_user_attributes"`
+	AllowSubdomains                  bool     `json:"allow_subdomains"`
+	AllowIdpInitiated                bool     `json:"allow_idp_initiated"`
```
Should `IDP` be capitalized?

kduprey replied:
This is currently the same name as what is in our [main Go repo](https://github.com/clerk/clerk_go/blob/kenton/user-2719-add-user-object-to-sessioncreated-webhook/api/serialize/saml_account.go#L36-L37).

kduprey replied:
Updated

Commit:
feat: add missing User struct fields

- Add `EnterpriseAccount` type with SAML/OAuth enterprise connection support
- Enhance `User` struct with new fields:
  - `CreatedAt`, `UpdatedAt`, `MFAEnabledAt`, `MFADisabledAt`, `LegalAcceptedAt`
  - `EnterpriseAccounts` for enterprise SSO support
- Update `EmailAddress` with `MatchesSsoConnection` field
- Add shared `Verification` and `LinkedIdentification` types

Signed-off-by: Kenton Duprey <kenton@clerk.dev>

Commit:
refactor: clean up deprecated Domain fields in enterpriseAccountConnection and samlAccountConnection structs

Signed-off-by: Kenton Duprey <kenton@clerk.dev>

Commit:
fix: correct casing for MatchesSSOConnection and AllowIDPInitiated fields

Signed-off-by: Kenton Duprey <kenton@clerk.dev>

jfoshee reviewed (approved):
Thanks!

Commit:
feat: add message field to Verification struct and username field to OrganizationMembershipPublicUserData struct

Signed-off-by: Kenton Duprey <kenton@clerk.dev>

Commit:
feat: export EnterpriseAccountConnection, SAMLAccountConnection types and add ExternalAccount.PhoneNumber

- Export `EnterpriseAccountConnection` type to make it publicly accessible
- Add `PhoneNumber` field to `ExternalAccount` struct
- Update `SAMLConnection` field references to use exported type

Signed-off-by: Kenton Duprey <kenton@clerk.dev>

jfoshee reviewed (approved):
LGTM, Thanks
